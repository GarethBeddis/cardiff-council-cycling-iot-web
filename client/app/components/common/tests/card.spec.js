import React from 'react'
import { shallow } from 'enzyme'

import Card from '../card'

// Prepare a mock function
const mockOnClick = jest.fn()

describe('Card Component', () => {
  // Reset the Mock
  beforeEach(() => {
    mockOnClick.mockClear()
  })
  // General Tests
  it('Renders without crashing', () => {
    shallow(<Card>Test Child</Card>)
  })
  it('OnClick event fires', () => {
    let wrapper = shallow(<Card onClick={mockOnClick}>Test Child</Card>)
    wrapper.find('a.card').first().simulate('click')
    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })
  // Snapshot Tests
  describe('Snapshots', () => {
    it('Standard card style matches Snapshot', () => {
      let wrapper = shallow(<Card>Test Child</Card>)
      expect(wrapper).toMatchSnapshot()
    })
    it('OnClick card style matches Snapshot', () => {
      let wrapper = shallow(<Card onClick={mockOnClick}>Test Child</Card>)
      expect(wrapper).toMatchSnapshot()
    })
    it('Link card style matches Snapshot', () => {
      let wrapper = shallow(<Card link='/test'>Test Child</Card>)
      expect(wrapper).toMatchSnapshot()
    })
    it('Custom class card style matches Snapshot', () => {
      let wrapper = shallow(<Card className='test'>Test Child</Card>)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
