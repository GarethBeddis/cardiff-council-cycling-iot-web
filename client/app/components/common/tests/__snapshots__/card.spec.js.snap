// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Snapshots Custom class card style matches Snapshot 1`] = `
<div
  className="card  test"
>
  Test Child
</div>
`;

exports[`Card Component Snapshots Link card style matches Snapshot 1`] = `
<Link
  className="card clickable"
  replace={false}
  to="/test"
>
  Test Child
</Link>
`;

exports[`Card Component Snapshots OnClick card style matches Snapshot 1`] = `
<a
  className="card clickable"
  onClick={[MockFunction]}
>
  Test Child
</a>
`;

exports[`Card Component Snapshots Standard card style matches Snapshot 1`] = `
<div
  className="card "
>
  Test Child
</div>
`;
